src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2220 @@
+
+    """
+    # Dict where key is the version of xtadapter package and value is information about the package
+    version_package = {}
+    # Package is named as f"delmic-{adapter}-{bitness}bit-v{version}"
+    # delmic-xtadapter-32bit-v1.11.2-dev
+    # delmic-xtadapter-32bit-v1.11.2.zip
+    # delmic-fastem-xtadapter-64bit-v1.11.2-dev
+    regex = r"delmic-([a-zA-Z-]+)-(\d+)bit-v([\d.]+)"
+    if is_zip:
+        regex += r".*.zip"
+    if os.path.isdir(directory):
+        for entity in os.listdir(directory):
+            result = re.search(regex, entity)
+            entity_path = os.path.join(directory, entity)
+            if result and (
+                entity.endswith(".zip") if is_zip
+                else os.path.isdir(entity_path) and len([entity for entity in os.listdir(entity_path) if entity.endswith(".exe")]) == 1
+            ):
+                package = Package()
+                package.adapter = result.group(1)
+                package.bitness = result.group(2) + "bit"
+                package.version = result.group(3)
+                package.name = entity
+                package.path = entity_path
+                # If not a zip file, assign the path of the executable
+                if not is_zip:
+                    for entity in os.listdir(package.path):
+                        if entity.endswith(".exe"):
+                            package.path = os.path.join(package.path, entity)
+                            break
+                # Filter using adapter type and bitness
+                if bitness == package.bitness and adapter == package.adapter:
+                    version_package[package.version] = package
+        # Check if any version is newer than the current one
+        versions = list(version_package)
+        if versions:
+            # Find the latest version
+            latest_version = max(versions, key=lambda s: [int(u) for u in s.split(".")])
+            if current_version < latest_version:
+                logging.info("The latest version is {}.\n".format(latest_version))
+                latest_package = version_package[latest_version]
+                if "-dev" in latest_package.name:
+                    logging.warning(
+                        "{} is a development version.\n".format(latest_package.name)
+                    )
+                return latest_package
+    return None
+
+
+# information on compatible versions
+debug_connection_info = f"""PYRO 5: {pkg_resources.get_distribution('Pyro5').version},
+msgpack-numpy: {pkg_resources.get_distribution('msgpack-numpy').version},
+msgpack_numpy_file: {msgpack_numpy.__file__},
+msgpack version: {pkg_resources.get_distribution('msgpack').version}
+msgpack_file: {msgpack.__file__}
+
+This is likely a dependencies issue:
+make sure you have the following matching versions of msgpack and msgpack-numpy installed:
+msgpack==0.5.6, msgpack-numpy==0.4.4
+or
+msgpack==1.0.3 msgpack-numpy==0.4.8
+"""
+
+class FIBSEM(model.HwComponent):
```

Comment:
Why not just name it `SEM`, this is not specific for FIBSEMs or is it? At least I think it would be nice to have autoscript support for regular SEMs as well

Reply:
Changed to SEM to be consistent with other driver, added a comment explaining what is supported